### ISSUE TITLE
Investigate using Appveyor for CI instead of Travis

### ISSUE THREAD

**Issue by JasonShin** [machinelearnjs/machinelearnjs#87]
* **I'm submitting a ...**

[/] enhancement

* **Summary**

Due to the demand for supporting Windows for the development and the production environment, it would be nice to have a Windows CI environment, but it is not supported by Travis CI. After conducting an initial investigation, [Appveyor](https://www.appveyor.com/) stands out as an alternative to Travis since it is free for open source and provides both Windows and Linux build environments. The ticket aims to do a further investigation and possibly make a pull request with Appveyor CI config file.

**Comment by LSBOSS** [comment]
Hi @JasonShin, 

I'm currently playing around with Appveyor in combination with my local fork of Kalimdor.js.
The Windows version of it looks already pretty nice (and is doing the same building/testing scripts as the Travis script):

https://ci.appveyor.com/project/LSBOSS/kalimdorjs

I have some questions/remarks, though: 

### Node version
In the current Travis script I see that the CI should run for Node 6, 7, 8, and 9. Here's the thing: 

- Yarn support for 7 is unsupported and my produce bugs. 
- Moreover the package vuepress@0.12.0 requires node to be >= 8. That's why building with Node 6 fails 
- Also fs-extra@7.0.0 is not supporting Node 7. That's why building with Node 7 fails. 

Giving that Node 7 and 9 are already in End of Life and will not receive any support in the future and Node 6's maintenance support ends April 2019 I`ll suggest to focus on the more current LTS and maybe the bleeding edge version of Node (that is 8, 10 and 11). What do you think? 

More info: https://github.com/nodejs/Release#release-schedule 

### Version & Deploying 
Releasing a new version of Kalimdor.js with the help of release-it also bumps the version in package.json. I would like to keep the version inside of Appveyor in sync with the "official" one in package.json. This might not be needed as currently the CIs (both Travis and Appveyor) are only used for build validation. 

OR

Once the Linux build validation is in place and Appveyor is replacing Travis we can experiment with Appveyor being used for deployment (i.e. increasing version, pushing to npm etc.). We could say, that releasing a new version is as easy as just tagging a certain commit and push that tag to the repository. 

What do you think of that?

**Comment by JasonShin** [comment]
Hey @LSBOSS, I really appreciate your inputs, it is always great to collaborate with cool and smart people!

**Node Version**

I think we are already on the same page with this! In fact, other collaborators and I have agreed to bump up the minimum Node version requirement to `>=8.11.0` because we _"probably"_  want to use [Tensorflow.js's Linear Algebra API](https://js.tensorflow.org/) to refactor the codebase. As you have suggested, I am happy to scope it down to the bleeding edge versions that you have listed above.

Also, feel free to join https://t.me/joinchat/Hj2nWgsslUN0BWGi0CEOHg to meet with the other contributors ;)

**Version & Deploying**

I love the idea of automated deployment through the CI environment, and I think it is necessary to support robust continuous deployments to deal with bug hotfixes and etc. However, can we please formulate concrete plans before we go ahead with it? For example

1. In what conditions does the CI know to make a release? Is every git merge to master going to be released? It would be nice to have a flag to tell the CI not to release in certain circumstances, for instance, updating the README.md

2. Devising disaster recovery plans; although running the `release-it` will be automated by CI, triggering the process will still be done by us (I think), which still leaves a room for making mistakes.

3. `releasing a new version is as easy as just tagging a certain commit and push that tag to the repository.` please explain more about this =) 

In short, I think `1. Node version` is fine and I think `2. Version & Deploying` would need some planning beforehand.

**Comment by LSBOSS** [comment]
I added a PR with a working Appveyor script here: https://github.com/kalimdorjs/kalimdorjs/pull/125

**Comment by LSBOSS** [comment]
@Release: 

Usually this is handled in either of two ways:
- You could let the release process only be triggered via specific tags (e.g. Release-1.01.2345). So it's under your control when a commit is tagged and you can decided if minor updates to a readme file is worth an Release. Also, AFAIK, only the repo owner can tag commits. So releasing a new version of Kalimdor.JS will stay under your control as well
- You could create a Release branch in which you merge from master from time to time. Every commit/merge in this Release branch is counting as "Release" which triggers the whole release pipeline (version increment, testing, building, packaging, push to npm etc.)
